cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java

```diff
@@ -250,6 +277,49 @@ public Service generateService() {
             volumeList.add(createSecretVolume(passwordSecret.getSecretName(), passwordSecret.getSecretName(), isOpenShift));
         }
 
+        volumeList.addAll(getExternalConfigurationVolumes(isOpenShift));
+
+        return volumeList;
+    }
+
+    private List<Volume> getExternalConfigurationVolumes(boolean isOpenShift)  {
+        int mode = 0444;
+        if (isOpenShift) {
+            mode = 0440;
+        }
+
+        List<Volume> volumeList = new ArrayList<>(0);
+
+        for (ExternalConfigurationVolumeSource volume : externalVolumes)    {
+            String name = volume.getName();
+
+            if (name != null) {
+                if (volume.getConfigMap() != null && volume.getSecret() == null) {
+                    ConfigMapVolumeSource source = volume.getConfigMap();
+                    source.setDefaultMode(mode);
+
+                    Volume newVol = new VolumeBuilder()
+                            .withName(EXTERNAL_CONFIGURATION_VOLUME_NAME_PREFIX + name)
+                            .withConfigMap(source)
+                            .build();
+
+                    volumeList.add(newVol);
+                } else if (volume.getConfigMap() == null && volume.getSecret() != null)    {
+                    SecretVolumeSource source = volume.getSecret();
+                    source.setDefaultMode(mode);
+
+                    Volume newVol = new VolumeBuilder()
+                            .withName(EXTERNAL_CONFIGURATION_VOLUME_NAME_PREFIX + name)
+                            .withSecret(source)
+                            .build();
+
+                    volumeList.add(newVol);
+                } else {
+                    log.warn("Volume {} with external Kafka Connect configuration has to contain exactly one volume source reference to either ConfigMap or Secret", name);
+                }
+            }
+        }
+
```

Comment:
No need to change but what do you think about the following? 

```java
if (volume.getConfigMap() != null && volume.getSecret() != null) {
    log.warn("Volume {} with external Kafka Connect configuration has to contain exactly one volume source reference to either ConfigMap or Secret", name);
} else {
    VolumeBuilder newVolBuilder = new VolumeBuilder()
                .withName(EXTERNAL_CONFIGURATION_VOLUME_NAME_PREFIX + name);

    if (volume.getConfigMap() != null) {
        ConfigMapVolumeSource source = volume.getConfigMap();
        source.setDefaultMode(mode);
    
        newVolBuilder.withConfigMap(source);
    } else if (volume.getSecret() != null) {
        SecretVolumeSource source = volume.getSecret();
        source.setDefaultMode(mode);

        newVolBuilder.withSecret(source);
    }

    volumeList.add(newVolBuilder.build());
}
```

Reply:
I improved the code a bit. I haven't used exactly the same code as you suggested because it doesn't address the situation when none of them is specified. But it is quite similar. Please have a look if you want.

Reply:
Right, we could add another `if` to check that at least one is not `null` in the `else` but I am fine with your latest changes.